airbyte-integrations/connectors/source-googleanalytics-singer/source_googleanalytics_singer/source.py

```diff
@@ -24,25 +24,33 @@
 
 import json
 import os
-import tempfile
 
 from airbyte_protocol import AirbyteConnectionStatus, Status
-from base_singer import SingerSource
+from base_singer import AirbyteLogger, SingerSource
 from tap_google_analytics import GAClient
 
-CREDENTIALS_FILE = os.path.join(tempfile.gettempdir(), "credentials.json")
```

Comment:
This tempfile is never deleted so it might cause problems as more and more sync are run using this source...

Instead, a new method on the source abstraction called `configure` allows source to override and add extra temp file to persist in the Temp_dir context...